### PR TITLE
Make embed_url modifier load embeds without cookies from Vimeo / YouTube

### PR DESCRIPTION
**Pull request by jwahdatehagh**
Youtube has a no-cookie feature when embedding videos via the `youtube-nocookie.com`-url. This way users are not tracked and reference to these cookies can be removed from privacy policies & cookie consent popovers.

Same goes for vimeo videos with the `dnt` query parameter set.

### CODE REVIEW

File: src/Modifiers/CoreModifiers.php [hunk 1]
```diff
@@ -2398,12 +2406,14 @@ public function embedUrl($url)
...
 
-            return $url;
+        if (Str::contains($url, 'youtube.com/watch?v=')) {
+            $url = str_replace('watch?v=', 'embed/', $url);
         }
 
-        if (Str::contains($url, 'vimeo')) {
-            return str_replace('/vimeo.com', '/player.vimeo.com/video', $url);
+        if (Str::contains($url, 'youtube.com')) {
+            $url = str_replace('youtube.com', 'youtube-nocookie.com', $url);
```
Comment (damcclean) [inline src/Modifiers/CoreModifiers.php:2416]: If you are doing the no cookie stuff for `youtube.com`, could you not also do it for `youtu.be`?

Comment (jwahdatehagh) [reply src/Modifiers/CoreModifiers.php:2416]: I think it's doing it for both cases 🙂

`youtu.be` is changed to `youtube.com` which then later is changed to `yotube-nocookie.com`.

This was the way with the least levels of indentation and repetition i could come up with 😅

Comment (damcclean) [reply src/Modifiers/CoreModifiers.php:2416]: Ah okay... I've not tested it out, just looked over the new code.

Comment (jwahdatehagh) [reply src/Modifiers/CoreModifiers.php:2416]: I could write a couple unit tests if that helps...

**Comment by jwahdatehagh** [comment]
Added the `EmbedUrlTest` @damcclean ...

**Comment by damcclean** [comment]
Nice! That looks good to me. Just need to wait until it gets reviewed by the gents!

**Comment by jasonvarga** [comment]
Neat! Had no idea about this.

Can't see a reason why you'd _not_ want to use youtube-nocookie.com.

**Comment by jasonvarga** [comment]
Maybe Cookie Monster.

**Comment by danijelk** [comment]
> Maybe Cookie Monster.

And google tag manager..
Can this be reverted and added as an option as it's a breaking change?

https://stackoverflow.com/questions/64973469/no-access-control-allow-origin-error-while-embeding-youtube-video

**Pull request merged**